File: Work/WK/MS/Lib/JxBackendServiceLibrary/JxBackendServiceStd/Model/ViewModel/Chat/LastMessageViewModel.cs
using JxBackendService.Common.Util;

namespace JxBackendService.Model.ViewModel.Chat
{
    public class LastMessageViewModel
    {
        /// <summary>房間號</summary>
        public string RoomID { get; set; }

        /// <summary>顯示名稱</summary>
        public string RoomName { get; set; }

        /// <summary>頭像網址</summary>
        public string AvatarUrl { get; set; }

        /// <summary>最後一則訊息ID</summary>
        public long MessageID { get; set; }

        public string MessageIDText => MessageID.ToString();

        public int MessageType { get; set; }

        /// <summary>最後一則訊息內容</summary>
        public string Message { get; set; }

        /// <summary>最後一則訊息發送時間</summary>
        public long PublishTimestamp { get; set; }

        /// <summary>最後一則訊息顯示文字時間</summary>
        public string PublishDateTimeText => PublishTimestamp.ToDateTime().ToFormatDateTimeString();

        /// <summary>未讀數量</summary>
        public int UnreadCount { get; set; }
    }
}

File: Work/WK/MS/Web_WebSV/Web/Web.Core/wwwroot/CTS/ClientApp/mimi-dating/src/models/LastMessageViewModel.ts
export interface LastMessageViewModel {
  //房間號
  roomID: string;

  //顯示名稱
  roomName: string;

  //頭像網址
  avatarUrl: string;

  messageID: number;

  //最後一則訊息ID
  messageIDText: string;

  messageType: number;

  //最後一則訊息內容
  message: string;

  //最後一則訊息發送時間
  publishTimestamp: number;

  //最後一則訊息顯示文字時間
  publishDateTimeText: string;

  //未讀數量
  unreadCount: number;
}
